src/cargo/ops/cargo_clean.rs

```diff
@@ -136,9 +136,13 @@ pub fn clean(ws: &Workspace<'_>, opts: &CleanOptions<'_>) -> CargoResult<()> {
     for pkg in packages {
         let pkg_dir = format!("{}-*", pkg.name());
 
-        // Clean fingerprints.
         for (_, layout) in &layouts_with_host {
+            // Clean fingerprints.
             rm_rf_glob(&layout.fingerprint().join(&pkg_dir), config)?;
+            // Clean target/doc.
+            rm_rf(&layout.doc(), config)?;
+            // Clean target/doc/src.
+            rm_rf(&layout.src(), config)?;
```

Comment:
This seems to delete the entire doc directory.  I believe this should only be deleting the directory for the specific package.